### ISSUE TITLE
Issues with preview images generation

### ISSUE THREAD

**Issue by coffeverton** [bookwyrm-social/bookwyrm#3106]
Hi!
I enabled the image preview generation (ENABLE_PREVIEW_IMAGES) but I'm having some issues.

When I ran the command to create the images (./bw-dev generate_preview_images --all), the font file is not being found. This message is show on the console:
`Just a sec, downloading Source Han Sans
Couldn't open font file /app/static/fonts/source_han_sans/SourceHanSans-VF.ttf.ttc for writing`
But I can see that the file exists:
` ls -l static/fonts/source_han_sans/
total 37200
-rw-r--r--. 1 root root 38090616 Nov 11 11:48 SourceHanSans-VF.ttf.ttc
`

It does generate the images, but with a different font:
https://deposito.velhaestante.com.br/images/previews/covers/11640-4b1aae93-6a75-4bfb-a65b-f5b36ab0e2c0.jpg

Also, the book pages are not using the generated previews. Instead, the instance logo is being used: https://velhaestante.com.br/book/54410/s/o-jogo-da-amarelinha

Is there anything else that I need to do? I set up the variable on the .env file, restart the instance, ran the command to generate the previews and sync the files to my S3 service (I'm using Cloudflare R2).

If I ran the collectstatic command, it shows the same error message about the font file not being found.

Any help is appreciated.

**Comment by Thovi98** [comment]
I think the issue here is that the file is set as `root:root`. Did you try with the same owner permissions as the rest of your Bookwyrm's directory?
For me it would be `chown bookwyrm:www-data [...]`

**Comment by coffeverton** [comment]
@Thovi98 I tried to set the permissions to the owner of the directory, but since I'm using Docker, I don't think that made a difference. But your comment helped me to look in the right direction.

When I tried to set the permissions inside the docker container, I found that the font's folder content inside the container was different from the content seen on the host.

Here is what I see from the host machine:
**ls -l static/fonts/source_han_sans/
total 37200
-rw-r--r--. 1 opc opc 38090616 Nov 11 11:48 SourceHanSans-VF.ttf.ttc**


And here from inside the bookwyrm-web container:
**ls -l static/fonts/source_han_sans/
total 37212
-rwxrwxrwx. 1 1000 1000     4463 Aug  7  2022 LICENSE.txt
-rwxrwxrwx. 1 1000 1000      497 Aug  7  2022 README.txt
-rw-r--r--. 1 root root 38090616 Dec  4 21:52 SourceHanSans-VF.ttf.ttc**

At first, the "SourceHanSans-VF.ttf.ttc" was not inside the container, so I just copied it to the right location (/var/lib/docker/volumes/bookwyrm_static_volume/_data/fonts/source_han_sans/) and now it works!
